.github/workflows/prebuild.yml

```diff
@@ -0,0 +1,83 @@
+name: Generate prebuild for develop branch
+
+on:
+  push:
+    branches: [develop]
+  workflow_dispatch:
+
+permissions:
+  contents: write
+
+env:
+  NODE_OPTIONS: --max-old-space-size=4096
+
+jobs:
+  reset-prebuild:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Reset prebuild branch to develop
+        uses: nicksnell/action-reset-repo@master
+        with:
+          base_branch: develop
+          reset_branch: prebuild
+
+  prebuild:
+    needs: reset-prebuild
+    timeout-minutes: 20
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: develop
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.11"
+
+      - name: Hash frontend source code
+        id: hash-frontend
+        run: |
+          python tools/frontend/hash_source.py
+          echo "HASH=$(cat hash.txt)" >> $GITHUB_OUTPUT
+          rm hash.txt
+        shell: bash
+
+      - name: Restore cached frontend build
+        id: cache-fe-build
+        uses: actions/cache@v4
+        with:
+          path: |
+            taipy/gui/webapp
+            taipy/gui_core/lib
+          key: frontend-build-${{ runner.os }}-${{ steps.hash-frontend.outputs.HASH }}
+
+      - name: Setup node ${{ matrix.node-version }} on ${{ matrix.os }}
+        if: steps.cache-fe-build.outputs.cache-hit != 'true'
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ matrix.node-version }}
```

Comment:
I don't see any matrix defined and I suppose we don't need one here ?

Reply:
oops, thanks for finding it! you are the best!